dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,144 @@
+from typing import List
+import numpy
+from dfastmi.kernel.typehints import Vector
+
+
+class BedLevelCalculator:
+    """
+    This class is used to calculate bed level related values.
+    """
+
+    def __init__(self, number_of_periods: int):
```

Comment:
In python you can specify that you expect `number_of_periods` to be an `int`, however, in python it is merely a suggestion :D

If you want to fail fast, you could add a check if the `number_of_periods` is indeed an `int`. Currently, if a non-integer is provided, it will probably crash at a later point with potentially a more vague error message.

Reply:
That is currently what I am doing with typehinting right?

Any suggestion on how to tackle the second part?

Reply:
discussed: assert (maybe?) or if raise when number_of_periods is not an int.

Reply:
added